src/main/java/edu/ucsb/cs156/happiercows/controllers/AnnouncementsController.java

```diff
@@ -133,54 +142,70 @@ public ResponseEntity<Object> getAnnouncementById(@Parameter(description = "The
             log.info("Start date not specified. Defaulting to current date.");
-            startDate = new Date();
+            updated.setStartDate(new Date());
         }
-
-        if (endDate != null && startDate.after(endDate)) {
-            return ResponseEntity.badRequest().body("Start date must be before end date.");
+        else {
+            updated.setStartDate(params.getStartDate());
         }
 
-        Optional<Announcement> announcementLookup = announcementRepository.findByAnnouncementId(id);
+        // ...But a start date after an end date is not OK.
+        // This error only occurs if the user provides an end date.
+        // Otherwise, the end date is NULL and the start date could be any date.
+        if(params.getEndDate() != null) {
+            if (updated.getStartDate().after(params.getEndDate())) {
+                throw new IllegalArgumentException("The start date may not be after the end date.");
+            }
+            else {
+                updated.setEndDate(params.getEndDate());
+            }
+        }
+        
 
-        if (!announcementLookup.isPresent()) {
-            return ResponseEntity.badRequest().body("Announcement could not be found. Invalid id.");
+        // The provided announcement text must not be empty or missing.
+        if (params.getAnnouncementText() == null || params.getAnnouncementText().equals("")) {
+            throw new IllegalArgumentException("Announcement Text cannot be empty.");
+        }
+        else {
+            updated.setAnnouncementText(params.getAnnouncementText());
         }
 
-        // Create the announcement
-        Announcement announcementObj = announcementLookup.get();
-        announcementObj.setStartDate(startDate);
-        announcementObj.setEndDate(endDate);
-        announcementObj.setAnnouncementText(announcementText);
+        // At this point we know that the announcement exists. So we can 
+        // also check if the user is in the commons at all. If not, the
+        // request is forbidden
+        User user = getCurrentUser().getUser();
+        Long userId = user.getId();
+        Long commonsId = updated.getCommonsId();
+        // Make sure the user is part of the commons and showChat is true, or user is an admin
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (!auth.getAuthorities().stream().anyMatch(a -> a.getAuthority().equals("ROLE_ADMIN"))){
+            log.info("User is not an admin, checking if they're in the relevant commons");
+            Optional<UserCommons> userCommonsLookup = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId);
 
-        // Save the announcement
-        announcementRepository.save(announcementObj);
-        return ResponseEntity.ok(announcementObj);
+            if (!userCommonsLookup.isPresent()) {
+                return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+            }
+        }
```

Comment:
Might not need this anymore if we change PUT to ADMIN only?

Also the comment about `showChat` is maybe a copy/paste error?

Reply:
Yes, I'll change the comment / simplify